pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/MutableDictionaryReader.java

```diff
@@ -186,4 +202,63 @@ public void print() {
   public boolean isEmpty() {
     return dictionaryIdBiMap.isEmpty();
   }
+
+  /**
+   * Checks if the given value lies within one of the partition ranges. If the value is not within
+   * any of the partition ranges, partitioning is dropped. The effect of that is:
+   * <ul>
+   *   <li> Subsequent values will not be checked against partition ranges. </li>
+   *   <li> Partition information will not be written out to the metadata for this column. </li>
+   * </ul>
+   *
+   * @param value Column value to check.
+   */
+  protected void checkPartition(Object value) {
+    if (partitionFunction != null) {
+       int partition = partitionFunction.getPartition(value);
+
+      for (IntRange partitionValue : partitionValues) {
+        if (partition >= partitionValue.getMinimumInteger() && partition <= partitionValue.getMaximumInteger()) {
+          return;
+        }
+      }
+
+      LOGGER.warn("Column '{}' not partitioned as specified for value: '{}', dropping partition metadata.", column,
```

Comment:
if one value falls out of partition, will we be printing this message for every row thereafter? 
Also, wil we be printing this message for use cases that do not have a partition funciton?
Isnt it better to add a null chekc up front and return right away? This code is in the consumption path, right?